pymaster/field.py

```diff
@@ -658,3 +685,279 @@ def get_ell_sampling(self):
         """
         ells = lib.get_ell_sampling_flat(self.fl, int(self.fl.fs.n_ell))
         return ells
+
+
+class NmtFieldCatalog(NmtField):
+    """ An :obj:`NmtFieldCatalog` object contains all the information
+    describing a given field sampled at the discrete positions of
+    a given catalog of sources. It can also be used for clustering studies,
+    in which the density of points itself is the field of interest.
+
+    Args:
+        positions (`array`): Source positions, provided as a list or array
+            of 2 arrays. If ``lonlat`` is True, the arrays should contain the
+            longitude and latitude of the sources, in this order, and in
+            degrees (e.g. R.A. and Dec. if using Equatorial coordinates).
+            Otherwise, the arrays should contain the colatitude and
+            longitude of each source in radians (i.e. the spherical
+            coordinates :math:`(\\theta,\\phi)`).
+        weights (`array`): An array containing the weight assigned to
+            each source.
+        field (`array`): A list of 1 or 2 arrays (for spin-0 and spin-s
+            fields, respectively), containing the value of the field
+            whose power spectra we aim to calculate at the positions of
+            each source.
+        lmax (:obj:`int`): Maximum multipole up to which the spherical
+            harmonics of this field will be computed.
+        lmax_mask (:obj:`int`): Maximum multipole up to which the spherical
+            harmonics of this field's mask will be computed. If negative,
+            it will default to ``4*lmax``.
+        spin (:obj:`int`): Spin of this field. If ``None`` it will
+            default to 0 or 2 if ``field`` contains 1 or 2 arrays,
+            respectively.
+        beam (`array`): Spherical harmonic transform of the instrumental beam
+            (assumed to be rotationally symmetric - i.e. no :math:`m`
+            dependence) associated with this field. If ``None``, no beam will
+            be corrected for. Otherwise, this array should have at least of
+            size ``lmax+1``.
```

Comment:
It is not clear to me why we would want to keep the beam as an argument to `NmtFieldCatalog` but not `NmtFieldCatalogClustering`

Reply:
By definition there should be no beam for the clustering catalog, right?
In the `NmtFieldCatalog` case, you are sampling a true sky map at the position of some sources, and that map may have been smoothed with some beam. For `NmtFieldCatalogClustering`, the quantity you are mapping is directly the density of points, so that's not something you can smooth.

Reply:
Fixed after discussing in person.